design/ceph/object/ceph-cosi-driver.md

```diff
@@ -0,0 +1,142 @@
+# Ceph COSI Driver Support
+
+## Targeted for v1.12
+
+## Background
+
+Container Object Storage Interface (COSI) is a specification for container orchestration frameworks to manage object storage. Even though there is no standard procotol defined for Object Store, it has flexibility to add support for all. The COSI spec abstracts common storage features such as create/delete buckets, grant/revoke access to buckets, attach/detach buckets, and more. COSI released v1alpha1 with Kubernetes 1.25. COSI is a new project and is not yet fully integrated by Kubernetes.
+More details about COSI can be found [here](https://kubernetes.io/blog/2022/09/02/cosi-kubernetes-object-storage-management/)
+It is projected that COSI will be the only supported object storage driver in the near future. In-tree drivers such as Ceph RGW will be replaced with their respective COSI drivers.
+
+## COSI Driver Deployment
+
+The [COSI controller](https://github.com/kubernetes-sigs/container-object-storage-interface-controller) is deployed as container in the default namespace. The Ceph COSI driver is deployed as a statefulset with a single replica along with [COSI sidecar container](https://github.com/kubernetes-sigs/container-object-storage-interface-provisioner-sidecar). The Ceph COSI driver can be deployed in any namespace not along with the COSI controller. The Ceph COSI driver is deployed with a service account that has the following RBAC permissions:
+
+## Integration plan with Rook
+
+The aim to support the v1alpha1 version of COSI in Rook v1.12. It will be extended to beta and release versions as appropriate. There should be option in `operator.yaml` to validate whether the COSI controller exists and another option at object store level to bring up the Ceph COSI driver. Each Ceph object store should have its own COSI driver.
+
+### Pre-requisites
+
+- COSI CRDs should be installed in the cluster via following command
+
+```bash
+kubectl apply -k github.com/kubernetes-sigs/container-object-storage-interface-api
+```
+
+- COSI controller should be deployed in the cluster via following command
+
+```bash
+kubectl apply -k github.com/kubernetes-sigs/container-object-storage-interface-controller
+```
+
+- Create Service Account and RBAC for COSI driver
+
+```bash
+kubectl create -f https://github.com/ceph/ceph-cosi/blob/master/resources/sa.yaml
+kubectl create -f https://github.com/ceph/ceph-cosi/blob/master/resources/rbac.yaml
```

Comment:
I'd like to consider appending these to rook's common.yaml, then we don't need those steps. The COSI driver is expected to become common, so it's nice to just include the rbac.